### ISSUE TITLE
Failed to upload

### ISSUE THREAD

**Issue by smitti** [nextcloud/deck#1121]
Uploaded files are commented on with Failed to upload. They are still there after a reload. The deletion does not work, the files remain on the card.

**To Reproduce**
1. Go to the Deck
2. Click on a card
3. Choose the file-upload-section
4. Choose the file-manager or drag-and-drop the file into the area to upload
5. The message "failed to upload" appears
6. after a reload of the browser-window the file appears
7. it´s impossible to delete the file

**Client details:**
 - Windows 10
 - Chrome, Firefox
 - Device: [Samsung Note 8, Chrome]

<details>
<summary>Server details</summary>

**Operating system**:

Hosteurope based 

MySQL

PHP 7.2

**Nextcloud version:** 16.0.3

**Where did you install Nextcloud from:** Official Download 

**Signing status:**

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.
```No errors have been found.

**List of activated apps:**
Accessibility	1.2.0	Offiziell	
Activity	2.9.1	Offiziell	
Announcement center	3.5.1	Offiziell	
AppOrder	0.7.1		
Auditing / Logging	1.6.0	Offiziell	
Brute-force settings	1.3.0	Offiziell	
Calendar	1.7.0	Offiziell	
Circles	0.17.4	Offiziell	
Collaborative tags	1.6.0	Offiziell	
Comments	1.6.0	Offiziell	
Contacts	3.1.3	Offiziell	
Deck	0.6.4		
Deleted files	1.6.0	Offiziell	
DropIt	0.2.1		
Federation	1.6.0	Offiziell	
File access control	1.6.0	Offiziell	
File sharing	1.8.0	Offiziell	
Files automated tagging	1.6.0	Offiziell	
Flow Upload	0.1.2		
Forms	1.0.3		
Gallery	18.3.0	Offiziell	
Group folders	4.0.3	Offiziell	
Impersonate	1.3.0	Offiziell	
JSLoader	1.0.2		
LDAP user and group backend	1.6.0	Offiziell	
Log Reader	2.1.0	Offiziell	
Mail	0.15.1	Offiziell	
Monitoring	1.6.0	Offiziell	
Nextcloud announcements	1.5.0	Offiziell	
Notes	3.0.0	Offiziell	
Notifications	2.4.1	Offiziell	
Notifications for calendar event updates	0.3.4		
OwnBackup	19.5.0		
Password policy	1.6.0	Offiziell	
PDF viewer	1.5.0	Offiziell	
Polls	0.10.2		
Quota warning	1.5.0		
Recommendations	0.4.0	Offiziell	
Right click	0.13.0		
SSO & SAML authentication	2.3.1	Offiziell	
Support	1.0.0	Offiziell	
Tasks	0.11.0		
Terms of service	1.2.2		
Text editor	2.8.0	Offiziell	
Theming	1.7.0	Offiziell	
Time Tracker	0.0.26		
Update notification	1.6.0	Offiziell	
Usage survey	1.4.0	Offiziell	
Versions	1.9.0	Offiziell	
Video player	1.5.0	Offiziell	
Viewer
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```<?php
$CONFIG = array (
  'instanceid' => 'oc3nijebcuf7',
  'passwordsalt' => 
  'secret' => 
  'trusted_domains' => 
  array (
    0 => 'www.b-u-p-daheim.de',
  ),
  'datadirectory' => '/is/htdocs/wp1,
  'dbtype' => 'mysql',
  'version' => '16.0.3.0',
  'overwrite.cli.url' => 'http://www.b-u-p-daheim.de',
  'dbname' => '',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'dbuser' => '',
  'dbpassword' => '',
  'installed' => true,
  'mail_from_address' => 'webmaster',
  'mail_smtpmode' => 'sendmail',
  'mail_sendmailmode' => 'pipe',
  'mail_domain' => 'b-u-p-daheim.de',
  'mail_smtphost' => '.server-he.de',
  'mail_smtpsecure' => 'ssl',
  'mail_smtpport' => '587',
  'app_install_overwrite' => 
  array (
    0 => 'admin_notifications',
  ),
  'ldapIgnoreNamingRules' => false,
  'ldapProviderFactory' => 'OCA\\User_LDAP\\LDAPProviderFactory',
  'has_rebuilt_cache' => true,
  'maintenance' => false,
  'theme' => '',
  'loglevel' => 2,
  'updater.secret' => '$2y$10$cNPnftzetElF/PM5r7fA7uQ7YoLAIO59AFQgK',
);


**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```{"reqId":"HXyd6QHT6ftjsLOMqzbq","level":4,"time":"2019-05-30T05:34:05+00:00","remoteAddr":"188.104.131.85","user":"Schmitt","app":"mail","method":"POST","url":"\/index.php\/apps\/mail\/api\/accounts","message":"Creating account failed: Server denied authentication.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/74.0.3729.169 Safari\/537.36","version":"16.0.1.1"}
{"reqId":"P5bpqx6aQ2gNDduDwOxC","level":4,"time":"2019-05-30T05:34:39+00:00","remoteAddr":"188.104.131.85","user":"Schmitt","app":"mail","method":"POST","url":"\/index.php\/apps\/mail\/api\/accounts","message":"Creating account failed: Server denied authentication.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/74.0.3729.169 Safari\/537.36","version":"16.0.1.1"}
{"reqId":"rRhnrpjKAuKbyt24zr8K","level":4,"time":"2019-05-30T05:34:42+00:00","remoteAddr":"188.104.131.85","user":"Schmitt","app":"mail","method":"POST","url":"\/index.php\/apps\/mail\/api\/accounts","message":"Creating account failed: Server denied authentication.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/74.0.3729.169 Safari\/537.36","version":"16.0.1.1"}
{"reqId":"ATwsqwR9nMzoSxz3pMwV","level":4,"time":"2019-05-30T05:35:25+00:00","remoteAddr":"188.104.131.85","user":"Schmitt","app":"mail","method":"POST","url":"\/index.php\/apps\/mail\/api\/accounts","message":"Creating account failed: Server denied authentication.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/74.0.3729.169 Safari\/537.36","version":"16.0.1.1"}
{"reqId":"sftP9Any2zoCFPAsQlzg","level":4,"time":"2019-05-30T05:35:28+00:00","remoteAddr":"188.104.131.85","user":"Schmitt","app":"mail","method":"POST","url":"\/index.php\/apps\/mail\/api\/accounts","message":"Creating account failed: Server denied authentication.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/74.0.3729.169 Safari\/537.36","version":"16.0.1.1"}
{"reqId":"jK4dHkAm3hiPD8Iwa7Um","level":4,"time":"2019-05-30T05:35:33+00:00","remoteAddr":"188.104.131.85","user":"Schmitt","app":"mail","method":"POST","url":"\/index.php\/apps\/mail\/api\/accounts","message":"Creating account failed: Server denied authentication.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) 


[index] Error: Error: Call to a member function getId() on array at <<closure>>

 0. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/apps/deck/lib/Activity/ActivityManager.php line 243
    OCA\Deck\Activity\ActivityManager->createEvent("deck_card", OCA\Deck\Db\Attachment {id: 46}, "attachment_delete", [], null)
 1. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/apps/deck/lib/Service/AttachmentService.php line 332
    OCA\Deck\Activity\ActivityManager->triggerEvent("deck_card", OCA\Deck\Db\Attachment {id: 46}, "attachment_delete")
 2. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/apps/deck/lib/Controller/AttachmentController.php line 82
    OCA\Deck\Service\AttachmentService->delete("230", "46")
 3. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/lib/private/AppFramework/Http/Dispatcher.php line 166
    OCA\Deck\Controller\AttachmentController->delete("230", "46")
 4. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/lib/private/AppFramework/Http/Dispatcher.php line 99
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Deck\Control ... {}, "delete")
 5. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/lib/private/AppFramework/App.php line 126
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Deck\Control ... {}, "delete")
 6. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
    OC\AppFramework\App::main("OCA\\Deck\\Cont ... r", "delete", OC\AppFramework\ ... {}, {cardId: "230",a ... "})
 7. <<closure>>
    OC\AppFramework\Routing\RouteActionHandler->__invoke({cardId: "230",a ... "})
 8. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/lib/private/Route/Router.php line 297
    undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {cardId: "230",a ... "})
 9. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/lib/base.php line 975
    OC\Route\Router->match("/apps/deck/cards/230/attachment/46")
10. /is/htdocs/wp11141389_5VUUJ7N9W6/b-u-p-daheim.de/index.php line 42
    OC::handleRequest()

DELETE /index.php/apps/deck/cards/230/attachment/46
from 188.105.81.148 by Schmitt at 2019-07-09T18:03:03+00:00
--





#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by putt1ck** [comment]
Confirmed here on NC16.01, Deck 6.04.

**Comment by juliushaertl** [comment]
Fix from #1117 was released with 0.6.5
